gui/wxpython/lmgr/frame.py

```diff
@@ -1115,28 +1062,13 @@ def ChangeLocation(self, location, mapset, dbase=None):
 
     def OnCreateMapset(self, event):
         """Create new mapset"""
-        dlg = wx.TextEntryDialog(parent=self,
-                                 message=_('Enter name for new mapset:'),
-                                 caption=_('Create new mapset'))
-
-        if dlg.ShowModal() == wx.ID_OK:
-            mapset = dlg.GetValue()
-            if not mapset:
-                GError(parent=self,
-                       message=_("No mapset provided. Operation canceled."))
-                return
+        db_node, loc_node, mapset_node = self.datacatalog.tree.GetCurrentDbLocationMapsetNode()
+        self.datacatalog.tree.CreateMapset(db_node, loc_node)
```

Comment:
This should call create_mapset_interactively directly, not through datacatalog. As for the reloading of datacatalog, for now just call self.datacatalog.tree.ReloadTreeItems() and we will later change this using signals.